validation/OpenSpaceToolkit/Astrodynamics/CrossValidator.test.cpp

```diff
@@ -0,0 +1,207 @@
+
+using ostk::core::ctnr::Array;
+using ostk::core::ctnr::Table;
+using ostk::core::filesystem::Path;
+using ostk::core::filesystem::File;
+using ostk::core::types::Integer;
+using ostk::core::types::Real;
+using ostk::core::types::Shared;
+using ostk::core::types::Size;
+using ostk::core::types::String;
+
+using ostk::math::object::VectorXd;
+
+using ostk::physics::coord::Frame;
+using ostk::physics::time::Instant;
+using ostk::physics::time::Scale;
+using ostk::physics::time::DateTime;
+using ostk::physics::coord::Position;
+using ostk::physics::coord::Velocity;
+
+using ostk::astro::trajectory::State;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+using ostk::astro::trajectory::state::CoordinatesBroker;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianPosition;
+using ostk::astro::trajectory::state::coordinatessubsets::CartesianVelocity;
+
+class OpenSpaceToolkit_Astrodynamics_Validation_CrossValidator : public ::testing::Test
+{
+   protected:
+    const Table table_ = Table::Load(
+        File::Path(Path::Parse({"/app/validation/OpenSpaceToolkit/Astrodynamics/data/test/test_table.csv"})),
+        Table::Format::CSV,
+        true
+    );
+
+    const ToolComparison toolComparisonGMAT_ = {
+        Tool::GMAT,
+        {
+            {Quantity::CARTESIAN_POSITION_GCRF, 1.0e-0},
+            {Quantity::CARTESIAN_VELOCITY_GCRF, 1.0e-3},
+            {Quantity::CARTESIAN_ACCELERATION_GCRF, 1.0e-3},
+            {Quantity::MASS, 1.0e-6},
+            {Quantity::MANEUVER_ACCELERATION_J2000, 1.0e-3},
+        },
+    };
+    const ToolComparison toolComparisonOREKIT_ = {
+        Tool::OREKIT,
+        {
+            {Quantity::CARTESIAN_POSITION_GCRF, 1.0e-0},
+            {Quantity::CARTESIAN_VELOCITY_GCRF, 1.0e-3},
+            {Quantity::CARTESIAN_ACCELERATION_GCRF, 1.0e-3},
+            {Quantity::MASS, 1.0e-6},
+            {Quantity::MANEUVER_ACCELERATION_J2000, 1.0e-3},
+        },
+    };
+    const ToolComparison toolComparisonOREKIT_PosVelOnly_ = {
+        Tool::OREKIT,
+        {
+            {Quantity::CARTESIAN_POSITION_GCRF, 1.0e-0},
+            {Quantity::CARTESIAN_VELOCITY_GCRF, 1.0e-3},
+        },
+    };
+
+    const Shared<CoordinatesBroker> coordinatesBrokerSPtr_ = std::make_shared<CoordinatesBroker>(CoordinatesBroker({
+        CartesianPosition::Default(),
+        CartesianVelocity::Default(),
+        CoordinatesSubset::Mass(),
+    }));
+
+    State defaultState_ = State::Undefined();
+    Array<VectorXd> referenceCoordinates_ = Array<VectorXd>::Empty();
+
+    void SetUp() override
+    {
+        VectorXd initialCoordinates(7);
+        initialCoordinates << 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 10.0;
+        defaultState_ = {Instant::J2000(), initialCoordinates, Frame::GCRF(), coordinatesBrokerSPtr_};
+
+        referenceCoordinates_ = {
+            initialCoordinates.segment(0, 3), initialCoordinates.segment(3, 3), initialCoordinates.segment(6, 1)
+        };
+    }
```

Comment:
(nit) To align with other test classes and classes in OSTk in general, the methods are usually added above the fields of the classes. Apply elsewhere